api_factory/generator/templates/$namespace/$name_$version/$service/client.py.j2

```diff
@@ -97,6 +100,38 @@ class {{ service.name }}:
             default_timeout=None,  # FIXME
             client_info=self.client_info,
         )
+        {%- if method.field_headers %}
+
+        # Ensure metadata is a mutable sequence.
+        if metadata is None:
```

Comment:
Since `metadata` defaults to empty-tuple, this check is not necessary; `list(metadata)` is sufficient.